src/endpoints/grpc/stream.h

```diff
@@ -139,8 +139,18 @@ namespace ccf::grpc
 
   template <typename T>
   static DetachedStreamPtr<T> detach_stream(
-    StreamPtr<T>&& stream, http::StreamOnCloseCallback close_cb = nullptr)
+    const std::shared_ptr<ccf::RpcContext>& ctx,
+    StreamPtr<T>&& stream,
+    http::StreamOnCloseCallback close_cb = nullptr)
   {
+    auto rpc_ctx_impl = dynamic_cast<ccf::RpcContextImpl*>(ctx.get());
+    if (rpc_ctx_impl == nullptr)
+    {
+      throw std::logic_error("Unexpected type for RpcContext");
+    }
+
+    rpc_ctx_impl->response_is_pending = true;
```

Comment:
Do we still need this if we keep `make_pending()`?

Reply:
Yes - `response_is_pending` is the actual indicator that the stream was detached and response shouldn't be sent immediately. `make_pending` is just the empty return type if we have no response status to set now.